util/py/scripts/bazel_aspect_tool_audit_sec_mmio_calls.py

```diff
@@ -0,0 +1,112 @@
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+"""Audit calls to sec_mmio_* functions within a single translation unit.
+
+This script is meant to be invoked by a Bazel aspect. To generate a report,
+users should run audit_sec_mmio_calls.py.
+
+For context, see <https://github.com/lowRISC/opentitan/issues/18634>.
+"""
+
+import subprocess
+import sys
+from pathlib import Path
+
+import clang.cindex
+
+from util.py.packages.lib.register_usage_report import CallSiteAnalyzer
+from util.py.packages.lib.register_usage_report import \
+    RegisterTokenPattern as TokPat
+from util.py.packages.lib.register_usage_report import RegisterUsageReportGroup
+
+# Define analyzers for each of the sec_mmio_* functions we care about. We give
+# each analyzer a list of token patterns where None is a stand-in for a register
+# offset.
+#
+# If the list of token patterns is incomplete, we will fail the Bazel build by
+# raising an uncaught exception.
+#
+# NOTE: This list is likely to become out of date as the codebase evolves.
+SEC_MMIO_FUNCTION_ANALYZERS = [
+    CallSiteAnalyzer(
+        function_name="sec_mmio_read32",
+        arg_index=0,
+        reg_token_patterns=[
+            TokPat(['0']),
+            TokPat(['4']),
+            TokPat(['8']),
```

Comment:
Do you mind adding a short comment explaining what these are for?

Reply:
So CallSiteAnalyzer requires that the list of token patterns is complete — if it encounters a call site that has no matching pattern, it fails the build. So `TokPat(['0'])` was necessary because the analyzer hit a call site equivalent to `sec_mmio_read32(0)`.

I think you were only asking about those single-digit patterns, right? I'll skip adding a comment because I was able to just delete these patterns. It turns out they're not needed anymore.